roaring/roaring.go

```diff
@@ -2529,6 +2841,28 @@ func unionBitmapBitmap(a, b *Container) *Container {
 	return output
 }
 
+// unions bitmap b into bitmap a, mutating a in place. The n value of
+// a will need to be repaired after the fact.
+func unionBitmapBitmapInPlace(a, b *Container) {
+	// local variables added to prevent BCE checks in loop
+	// see https://go101.org/article/bounds-check-elimination.html
+
+	var (
+		ab = a.bitmap[:bitmapN]
+		bb = b.bitmap[:bitmapN]
+	)
+
+	// Manually unroll loop to make it a little faster.
+	// TODO(rartoul): Can probably make this a few x faster using
+	// SIMD instructions.
+	for i := 0; i < bitmapN; i += 4 {
+		ab[i] = ab[i] | bb[i]
```

Comment:
I think `|=` would be more expressive here.

I'm also not sure whether the loop unrolling is effective, but I don't suppose it hurts?

Reply:
I got like 15%ish gains when I unrolled it manually, although thats just on my laptop

Reply:
this is another one of those overly aggressive optimization that helps a little, but isn't world changing. Can leave it out of its too messy

Reply:
switched to `|=`